src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -100,6 +100,12 @@ internal static class AccessCheck
                 case SymbolKind.NamedType:
                     return IsNamedTypeAccessible((NamedTypeSymbol)symbol, within, ref useSiteDiagnostics, basesBeingResolved);
 
+                case SymbolKind.Alias:
+                    return IsSymbolAccessibleCore(((AliasSymbol)symbol).Target, within, null, out failedThroughTypeCheck, compilation, ref useSiteDiagnostics, basesBeingResolved);
```

Comment:
>((AliasSymbol)symbol).Target [](start = 50, length = 28)

Is the Target guaranteed to be not null in all scenarios?

Reply:
> Is the Target guaranteed to be not null in all scenarios?

The compilers do not expose alias symbols with null targets.

---
In reply to: [192574873](https://github.com/dotnet/roslyn/pull/27312#discussion_r192574873) [](ancestors = 192574873)

Reply:
> The compilers do not expose alias symbols with null targets.

Are they ever created internally with null target?

---
In reply to: [192882805](https://github.com/dotnet/roslyn/pull/27312#discussion_r192882805) [](ancestors = 192882805,192574873)